desktop-app/app/components/NavigationControls/index.js

```diff
@@ -58,7 +59,10 @@ class NavigationControls extends Component {
           <Tooltip title="Reload">
             <div
               className={commonStyles.flexAlignVerticalMiddle}
-              onClick={this.props.triggerNavigationReload}
+              onClick={() => {
+                notifyPermissionToHandleReloadOrNewAddress();
```

Comment:
@jjavierdguezas Should we move `notifyPermissionToHandleReloadOrNewAddress()` to the ADDRESS_CHANGE action handler instead of the individual components? 

That way we won't miss this when some other component does the address change in the future.

Reply:
yes, this is an ugly solution.  I used the `NEW_ADDRESS` action first, but for this feature you need to know when the address changed **before** the webview reloads and starts asking for permission.  After lots of testing, changes, and retries I settled on this because sometimes (go back and forward for example) the `NEW_ADDRESS` action has a long delay and as a result of that the component was handling new permission requests before find out about reloads, breaking the logic.  The (ugly) solution was to send a new and specific event as soon as possible when I know the address changed or the page reloads 😓. I really don't know how to avoid this, and yes any future change that triggers a page reload or address change needs to call `notifyPermissionToHandleReloadOrNewAddress()` function